script/Systems/Files/InitFiles.lua

```diff
@@ -217,26 +218,23 @@ function InitFiles:writeUserInits()
     local configPath = filepath .. filename .. filetype
     local backupConfigPath = filepath .. filename .. "_backup" .. filetype
 
-    -- Open the configuration file for reading
+    -- If a config file exists, open it so we can take a backup.
     local openedFileReadable = io.open(configPath, "r")
-    if not openedFileReadable then
-        Log.Warn("Cannot open configuration file for reading: %s", configPath)
-        return
-    end
-
-    -- Create a safety backup
-    local oldContent = openedFileReadable:read("*a")
-    openedFileReadable:close()
+    if openedFileReadable then
+        -- Create a safety backup
+        local oldContent = openedFileReadable:read("*a")
+        openedFileReadable:close()
+
+        local backupFile, backupFileErr = io.open(backupConfigPath, "w")
+        if not backupFile then
+            Log.Warn("Cannot open backup configuration file for writing: %s", backupFileErr)
```

Comment:
Question for you @IllustrisJack: Should we be skipping saving of the config if writing the backup fails? Or should we just skip taking a backup?

Reply:
hm good question. i´d assume if the backup fails you´d also fail in writing the config, so a general fail would make sense.

Reply:
Fair enough